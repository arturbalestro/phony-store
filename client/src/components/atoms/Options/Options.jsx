import React, { useState } from "react";
import { connect } from "react-redux";
import { deleteProduct } from "../../../actions";
import { StyledOptions } from "../../../styles/StyledOptions";
import { StyledButton } from "../../../styles/StyledButton";
import EditProductModal from "../../modals/EditProductModal/EditProductModal";

//Button to call the action.
export const Options = (props) => {
  const { product, deleteProduct } = props;
  const [editModal, setEditModal] = useState(false);
  const toggleEditModal = () => setEditModal(!editModal);

  return (
    <StyledOptions>
      <div className="button-wrapper">
        <StyledButton
          className="options"
          onClick={toggleEditModal}
          title="EDIT"
        >
          EDIT
        </StyledButton>
        <EditProductModal
          isOpen={editModal}
          toggle={toggleEditModal}
          currentProduct={product}
        />

        <StyledButton
          className="options"
          onClick={() => deleteProduct(product.id)}
          title="DELETE"
        >
          DELETE
        </StyledButton>
      </div>
    </StyledOptions>
  );
};

const mapStateToProps = (state) => {
  return {
    productId: state.productId,
  };
};

//Dispatching action to use as props
const mapDispatchToProps = {
  deleteProduct,
};

export default connect(mapStateToProps, mapDispatchToProps)(Options);
