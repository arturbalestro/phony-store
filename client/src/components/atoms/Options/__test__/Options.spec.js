// @flow
import Options from "../Options";
import React from "react";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";

const middlewares = [];
const mockStore = configureStore(middlewares);
const initialState = {};
const store = mockStore(initialState);

const mockedProduct = {
  name: "Sckhar",
  type: "Red",
  history: "The Red Product King of Arton",
};

test("<Options /> renders correctly", () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Options currentProduct={mockedProduct} />
      </Provider>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
