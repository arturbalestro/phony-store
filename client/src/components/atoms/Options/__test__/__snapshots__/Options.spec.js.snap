// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Options /> renders correctly 1`] = `
<div
  className="sc-AxjAm cYRYYw"
>
  <div
    className="button-wrapper"
  >
    <button
      className="sc-AxirZ ervvaP options"
      onClick={[Function]}
      title="EDIT"
    >
      EDIT
    </button>
    <div />
    <button
      className="sc-AxirZ ervvaP options"
      onClick={[Function]}
      title="DELETE"
    >
      DELETE
    </button>
  </div>
</div>
`;
